src/workerd/api/worker-rpc.c++

```diff
@@ -365,4 +485,35 @@ kj::Promise<WorkerInterface::CustomEvent::Result>
     .outcome = EventOutcome::OK
   };
 }
+
+// =======================================================================================
+
+jsg::Ref<WorkerEntrypoint> WorkerEntrypoint::constructor(
+    const v8::FunctionCallbackInfo<v8::Value>& args,
+    jsg::Ref<ExecutionContext> ctx, jsg::JsObject env) {
+  // HACK: We take `FunctionCallbackInfo` mostly so that we can set properties directly on
+  //   `This()`. There ought to be a better way to get access to `this` in a constructor.
+  //   We *also* delcare `ctx` and `env` params more explicitly just for the sake of type checking.
+  jsg::Lock& js = jsg::Lock::from(args.GetIsolate());
+
+  jsg::JsObject self(args.This());
+  self.set(js, "ctx", jsg::JsValue(args[0]));
+  self.set(js, "env", jsg::JsValue(args[1]));
```

Comment:
I'm wondering if these shouldn't be defined as `readonly` instance properties on the instance. I also wonder if these should be marked non-enumerable. The former being more important.

Reply:
Eh, I don't feel the need for these to be anything other than plain old properties as you'd get by assigning them in JS. They exist for the benefit of the subclass, so if the subclass decides to overwrite them, whatever.